Add dark theme

Aman-Codes opened the pull request:
<!--  Thanks for sending a pull request!  -->

## Proposed changes
Added the following enhancements in E2E dashboard:
- Added dark theme
- Removed status badge from all workflows
- Added rotating spinner for currently running jobs
- Removed build and release pipeline runs from all workflow runs
- Added new status badge

## Screenshots

![image](https://user-images.githubusercontent.com/54680709/140014416-d882b360-4e2c-42f2-9dfb-39dc51cfbeff.png)

![image](https://user-images.githubusercontent.com/54680709/140014371-2b88d7bc-59db-4a80-8c2d-e31888780fed.png)

![TableDark](https://user-images.githubusercontent.com/54680709/140011454-fd641df5-a6f7-4da1-96cd-8687cf673d03.PNG)

![2](https://user-images.githubusercontent.com/54680709/140011511-7eff83cd-f7bf-49a3-bfee-4f4f54628407.PNG)

![1](https://user-images.githubusercontent.com/54680709/140011514-ce7b481a-7e90-4197-83a5-bccd2e9bc67b.PNG)

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus-e2e/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] I have added necessary documentation (if appropriate)
- [x] I have added Screenshots or Terminal snippets in the PR comment to validate the successful execution of changes

**Documentation**
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] data-cy=* parameters have been added in codebase ( ChaosCenter Tests )
- [ ] data-cy=* parameters need to be added in codebase ( ChaosCenter Tests )
- [x] No updates required.

**Code Review**
- [ ] Does the test handle fatal exceptions, ie. uncaught exceptions from 3rd party library (if any)

**Issue**
- [ ] Tasks in issue are checked off

uditgaurav commented:
Instead of an empty spinner running on the table can we make the spinner of 4 dynamic? similar thing on steps as well.

![image](https://user-images.githubusercontent.com/35391335/140018042-9ec26e3b-9e5d-4d8d-b9d3-06c2e9453723.png)

Aman-Codes commented:
@uditgaurav Updated it to 
![image](https://user-images.githubusercontent.com/54680709/140635353-2d30cd91-638d-436d-9f13-7c7a325ca396.png)

Aman-Codes commented:
@uditgaurav Center aligned heading and reduced its font size
![image](https://user-images.githubusercontent.com/54680709/140916138-8ccc7639-b58f-4a50-8797-e7abd5fb32f3.png)

uditgaurav commented:
@Aman-Codes it would be good if we reduce the size of what we were having already. cc: @Jonsy13 @oumkale

Jonsy13 commented on src/containers/layouts/Scaffold/styles.js:
```diff
@@ -14,11 +14,11 @@ const useStyles = makeStyles((theme) => ({
       marginTop: theme.spacing(1),
-      webkitBoxShadow: `inset 0 0 8px ${theme.palette.common.black}`,
-      backgroundColor: theme.palette.border.main,
+      webkitBoxShadow: `inset 0 0 8px ${theme?.palette?.common?.black}`,
```
Why are we using `?` with theme, it should always be there right? Why are we checking if it is undefined?

oumkale commented on src/components/CustomCard/index.jsx:
```diff
@@ -12,47 +12,53 @@ import { Icon } from "litmus-ui";
 
 const statusBadge = (step) => {
-  const classes = useStyles();
   if (step?.status !== "completed") {
```
Hi @Aman-Codes,
Will you use a switch case here?

oumkale commented on src/components/Header/styles.js:
```diff
@@ -10,39 +10,23 @@ const useStyles = makeStyles((theme) => ({
     padding: theme.spacing(0, 7.5),
     "& *": {
-      color: theme.palette.text.secondary,
+      color: "white",
```
Use here `theme` color

oumkale commented:
@Aman-Codes Can you send a link so that we can have a better look?

Aman-Codes commented:
> @Aman-Codes Can you send a link so that we can have a better look?

https://aman-codes.github.io/litmus-e2e/dashboard/

oumkale commented:
Hi @Aman-Codes,

Make sure that all components are having proper theme in both themes.
![Screenshot from 2021-11-12 18-11-07](https://user-images.githubusercontent.com/34105373/141469469-91849ef5-87cc-4d8f-8ab4-9df1c29ac951.png)
![Screenshot from 2021-11-12 18-13-39](https://user-images.githubusercontent.com/34105373/141469476-2da019f8-a7c1-484f-bf10-3007d32828cb.png)
![Screenshot from 2021-11-12 18-15-09](https://user-images.githubusercontent.com/34105373/141469482-9eb78c38-4f69-4ffd-8754-043ff6c6e314.png)
![Screenshot from 2021-11-12 18-15-13](https://user-images.githubusercontent.com/34105373/141469488-238e1f2f-8df8-40f7-b1ba-1206d53f2da4.png)

Aman-Codes commented:
Updated the PR and deployed it to https://aman-codes.github.io/litmus-e2e/dashboard/

Aman-Codes commented:
The last commit adds the following enhancements:
- Removed underline from all hyperlink texts
- Added a loading spinner while fetching table data
- Updated test coverage to use radial progress chart
- Add hover effect on header text (home, nightly runs, manual runs)
- Fixed status chart background in dark theme
- Made subheading, header and dropdown outline grey in dark theme
